### ISSUE TITLE
Portal Productisation - Components cleanup

### ISSUE THREAD

**Issue by jocelynmutso** [digiexpress-io/digiexpress-parent#74]
We have reached the point where some cleanup needs to happen before we can continue with creating Plugins.  

The following tasks should be broken down into individual issues and linked to this parent issue via a checklist.

- [x] #80 
- [x] #73 
- [x] #78 
- [x] #69 
- [x] #84 
- [x] #101 

---

## ITEM 1: Topics.tsx

Topics is used here: 

1. `Secondary.tsx` (mobile)
2. `SecondaryTablet`
3. `HeaderDesktop` (desktop)

In HeaderDesktop.tsx, it is Imported and used like this 

```typescript
import Sipoo from '../components’;

<Sipoo.Topics variant='column’ …>

```

There are 3 variants, used in all 3 breakpoints. And there are lots of small naming issues here, making it hard to understand which of these are used where.  
 

**Task 1:**

- Create a new folder for Topics under `src/core` and move `Topics` there
- Rename the components to show their use cases. (TopicsColumn, TopicsTree, TopicsDrawer are not descriptive names...)
- Rename the variants to show their use cases: ‘mobile’ | ‘tablet’ | ‘desktop’
- Copy this new folder and renamed code as it is to the PluginContext folder like you've normally been doing
- It is now possible to use the existing desktop, tablet, or mobile plugin hooks

---

## ITEM 2: CancelFormDialog.tsx

CancelFormDialog was designed to work across all breakpoints, which is causing us this problem, because we can’t split it into individual breakpoints for Plugin hooks.

In the future, it is highly likely that we will need a customised version for each viewport.  

**Task 2:** 

Part 1: 

- Break this component into 3 variants (mobile, tablet, desktop).
- If mobile and tablet views are sharing this dialog, ignore the tablet variant.  Just declare mobile and desktop variants.
- Don't spend time styling this or cleaning it up right now -- that is a task for later.  Instead, create the variants and be sure that they are correctly displayed in each viewport as expected.  
- It would probably make sense that, in the mobile variant, the buttons are stacked vertically instead of side-by-side.

**Part 2: Do not start this before we discuss your completed CancelFormDialog with variants!**
- Create plugins as normal

---

## ITEM 3: RequireLogin.tsx

There is only one RequireLogin dialog, and it is used across all three viewports.

**Task 3**:

**TODO once ITEM 6: Refactoring AuthLink is complete**

## ITEM 4: Renaming NoticeBoard.tsx

Currently, NoticeBoard is only used in Desktop.   
 
**TODO once ITEM 6: Refactoring AuthLink is complete**

**Task 4:** 

- We should rename this to "NoticeBoardDesktop" so that it is 100% clear where it's used
- Be aware that it also uses `RequireLogin`.
- Create a Plugin from it like normal, but only for desktop.

## ITEM 5: TimestampFormatter

This one is an outlier.  It isn't exactly styled, and we need to use it across many places indiscriminately.  

It could also be that, in the future, we will discover more components like this.  This means that we need to add a new Plugin type --> `CommonPlugin`.

**Task 5:** 

Basically, you can start by creating the CommonPlugin interface in PluginContext/types.ts, and then adding it to Plugins.  A lot of stuff will break, but it's pretty much copy/paste from the existing patterns.

Something like this...

```typescript

//PluginContext/types.ts

export interface CommonPlugin {
 TimestampFormatter: React.FunctionComponent<TimestampFormatterProps>
}

export interface Plugins {
  desktop: DesktopPlugin,
  tablet: TabletPlugin,
  mobile: MobilePlugin,
  common: CommonPlugin
}

```
- Create a new hook in PluginContext/Context.tsx: `useCommonPlugin();`
- Create a `defaultCommonPlugin.ts` and just follow the pattern from the other files.
- `App/index.tsx`, you will add your default plugin type and then the overrides, like normal: 

```
  common: { ...Plugin.defaultCommonPlugin, TimestampFormatter }
```

You can use your hook in the same way as the others:

```
  const Plugins = Plugin.useCommon();
```

## ITEM 6: Refactoring AuthLink

Over time, Links.tsx and AuthLink have grown convoluted.  It is unclear, both from naming conventions and from unnecessarily complex usage of variants, exactly how AuthLink is used, and where it is used. 

This component needs to be cleaned up and made straightforward.

* Problem 1: It has a variant: `desktop` or undefined. This is not good. 
  * It should be `desktop`, `mobile`, or `tablet`. 
  * variant should always be defined.

* Problem 2: Does it even need a variant? There seems to be a `marginLeft` of 1 (linkStyle), which is the result of a variant. Is this even needed?

* Problem 3: There is also a `textVariant`. This shouldn’t even exist in here in the first place. 

**Task 6**: Solution

Analysis: 
* Where is the AuthLink is used
* How is the AuthLink used
* With what properties is it used

Move AuthLink to separate folder with its own variants if even needed.  

Come up with a better solution for `textVariant` .  One thing is clear: textVariant should be directly implemented inside of `Styles.Link`, with `dark` or `light` option.
